streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -56,32 +58,66 @@
                 final StateDirectory stateDirectory) {
         super(id, partitions, topology, consumer, true, stateMgr, stateDirectory, config);
 
-        closeTaskSensor = ThreadMetrics.closeTaskSensor(Thread.currentThread().getName(), metrics);
         processorContext = new StandbyContextImpl(id, config, stateMgr, metrics);
+        closeTaskSensor = ThreadMetrics.closeTaskSensor(Thread.currentThread().getName(), metrics);
+    }
+
+    @Override
+    public State state() {
+        return state;
+    }
+
+    @Override
+    public void transitionTo(final State newState) {
+        State.validateTransition(state, newState);
+        state = newState;
+    }
+
+    @Override
+    public void initializeIfNeeded() {
+        if (state() == State.CREATED) {
+            initializeMetadata();
+            initializeStateStores();
+
+            // no topology needs initialized, we can transit to RUNNING
+            // right after registered the stores
+            transitionTo(State.RUNNING);
```

Comment:
We could, but it seems to muddy the waters. It seems easier to think of "restoring" as some up-front setup, while "running" is the steady state. Since standbys don't need any up-front setup, it seems more straightforward to skip over restoring than to skip over running.